project/UI/CLIKit/ConsoleProgressReport.swift

```diff
@@ -0,0 +1,53 @@
+import NefModels
+import BowEffects
+import nef
+import Foundation
+
+public struct ConsoleProgressReport: ProgressReport {
+    public init() {}
+    
+    public func notify<E: Swift.Error, A>(_ event: ProgressEvent<A>) -> IO<E, Void> {
+        switch event.status {
+        
+        case .oneShot:
+            return ConsoleIO.print(event.step.progressDescription)
+            
+        case .inProgress:
+            return ConsoleIO.print(event.step.progressDescription, terminator: " ")
+            
+        case let .successful(info: info):
+            return ConsoleIO.print("✓".bold.green + info, terminator: "\n")
+            
+        case let .failed(error, info: info):
+            return ConsoleIO.print("✗".bold.red + info + error.localizedDescription, terminator: "\n")
+        }
+    }
+}
+
+extension ProgressReport {
+    func exit<A: CustomProgressDescription, E: Swift.Error>(success step: A) -> IO<E, Void> {
+        oneShot(step).map { Darwin.exit(0) }^
```

Comment:
to normalized messages, I think you can move here the `Outcome` starts
```suggestion
        oneShot("🙌".bold.green + step).map { Darwin.exit(0) }^
```